src/main/java/seedu/address/model/student/Student.java

```diff
@@ -25,7 +25,7 @@ public class Student {
 
     // Data fields
     private final Set<Tag> tags = new HashSet<>();
-    private final Optional<GroupName> groupName;
+    private Optional<GroupName> groupName;
```

Comment:
Does this impact the updating of UI?

I saw [this](https://github.com/nus-cs2103-AY2425S1/forum/issues/506) mentioned in the course repo. I think we can keep it as final and return a new Student object whenever we set their GroupName.